core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -66,18 +68,44 @@
    */
   static final int VALUE_SIZE_COPY_CUTOFF = 1 << 15;
 
+  /**
+   * Maximum size of a mutation (2GB).
+   */
+  static final long MAX_MUTATION_SIZE = 2l * (1 << 30);
+
   /**
    * Formats available for serializing Mutations. The formats are described in a <a href="doc-files/mutation-serialization.html">separate document</a>.
    */
   public static enum SERIALIZED_FORMAT {
     VERSION1, VERSION2
   }
 
+  final class MutationSize {
+    private int entries = 0;
+    private long sizeInBytes = 0;
+
+    private MutationSize() {}
+
+    private MutationSize(long rowLength) {
+      sizeInBytes = rowLength;
+    }
+
+    private MutationSize(int entries, long sizeInBytes) {
+      this.entries = entries;
+      this.sizeInBytes = sizeInBytes;
+    }
+
+    void update(long entrySize) {
```

Comment:
`update` could be renamed to `addEntryWithSize` or something else more descriptive.